libraries/lib-music-information-retrieval/MirUtils.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <algorithm>
+#include <cassert>
+#include <vector>
+
+namespace MIR
+{
+// Normal distribution parameters obtained by fitting a gaussian in the GTZAN
+// dataset tempo values.
+static constexpr auto bpmExpectedValue = 126.3333;
+
+// Function to generate numbers whose prime factorization contains only twos or
+// threes
+std::vector<int> GetPossibleBarDivisors(int lower, int upper);
+
+std::vector<int> GetPeakIndices(const std::vector<float>& odf);
+
+std::vector<float> GetNormalizedHann(int size);
+
+constexpr auto MapToPositiveHalfIndex(int index, int fullSize)
+{
+   assert(fullSize % 2 == 0);
```

Comment:
Make a `@pre` comment

Reply:
Is there also an intended assumption that `index < fullSize` ?

Reply:
Are you also assuming `fullSize > 0` ?  (Loops below won't terminate if it is 0)

Reply:
Making it unsigned doesn't protect against invalid input, since `fullSize > 0` is a pre-condition, so I don't see that much of a benefit.
But yes, I should handle invalid argument values.